### DISCUSSION TITLE
VCALENDAR Not Setting Time Correctly

### DISCUSSION THREAD

**Discussion by furick1** [code4recovery/12-step-meeting-list#1487] [General]
### Contact Details

furick1@gmail.com

### Website With Issue

https://recoverydharma.org/meetings/?meeting=rd-chicago-above-and-beyond-family-recovery-center

### What happened?

The add to calendar feature on meeting details seems to generate an incorrect time within the .ics file. When users try and add to calendar it isn't aligned with what shows on the website and is causing confusion. I have included some examples below using the latest plugin running on the Recovery Dharma website. The meeting timezone data is set at the meeting level with the global meeting timezone setting blank. The site is using the TSML UI plugin theme.

This issue was initially reported by user on Mac. Issue confirmed and calendar file examples generated by me on Windows in Virginia. @joshreisner I assume this is related to the recent timezone feature additions and the add to calendar function needs updated.

https://recoverydharma.org/meetings/?meeting=rd-chicago-above-and-beyond-family-recovery-center
Friday 1:00 PM – 2:00 PM EDT
(Friday 12:00 PM – 1:00 PM CDT)

BEGIN:VCALENDAR
VERSION:2.0
BEGIN:VEVENT
SUMMARY:RD Chicago @ Above and Beyond Family Recovery Center
DTSTAMP:20240809T170000Z
DTSTART;TZID=America/Chicago:20240809T130000
DTEND;TZID=America/Chicago:20240809T140000
LOCATION:Lake & Sacramento\n2942 W Lake St\, Chicago\, IL 60612\, USA
DESCRIPTION:Above and Beyond Family Recovery Center @ Lake & Sacramento\n2942 W. Lake. Chicago\, IL 60612
END:VEVENT
END:VCALENDAR

https://recoverydharma.org/meetings/?meeting=rd-wenatchee
Friday 3:00 PM – 4:00 PM EDT
(Friday 12:00 PM – 1:00 PM PDT)

BEGIN:VCALENDAR
VERSION:2.0
BEGIN:VEVENT
SUMMARY:RD Wenatchee
DTSTAMP:20240809T190000Z
DTSTART;TZID=America/Los_Angeles:20240809T150000
DTEND;TZID=America/Los_Angeles:20240809T160000
LOCATION:RD Wenatchee\n10 Orondo Ave Suite C\, Wenatchee\, WA 98801\, USA
DESCRIPTION:RD Wenatchee @\n10 Orondo Ave\, Suite C. Wenatchee WA 98801\n(within Wenatchee Community Acupuncture - Located inside the Mid Town Building. Entrance is in the alley.).\nMeeting with meditation\, reading + share.\nLocated inside the Mid Town Building. Entrance is in the alley.
END:VEVENT
END:VCALENDAR

### What browsers are you seeing the problem on?

Chrome

**Comment by gkovats** [comment]
Just confirming:
- RD Chicago shows as Central Timezone (CDT) on site, ICS and export
- Wenatchee shows as Pacific Timezone (PDT) on site, ICS and export

What's incorrect? I may be missing it

**Reply by furick1** [reply]
The timezone may show correctly but the time is off. Are you looking at the .ics dtstart and dtend values which are off compared to what shows on site?

**Comment by furick1** [comment]
From user:
![IMG_4891](https://github.com/user-attachments/assets/c9d20823-051c-43a0-b87f-aace5cfa76a8)
The info page says 5:00pm EST/2:00pm PDT but when I choose the option
 to add the event to my calendar it’s showing 8:00pm EST.

**Comment by furick1** [comment]
This is link to mtg of prior screenshot 

https://recoverydharma.org/meetings/?search=Women+/&meeting=women-non-binary-trans-only-7

Could this be a client device issue like incorrect timezone set in settings or in calendar app? When I look at the data it seems like the time is adjusted but the IANA in vcal is set to mtg location and not user IANA. It can stay the mtg IANA but the time shouldn’t convert then as it seems like there is a conversion by the plugin and again by calendar. Unfortunately my calendar on iPhone and windows doesn’t support ics or I need to configure so I haven’t tested fully end to end. The data looked wrong though which is why I posted what I thought looked like a bug.

**Comment by furick1** [comment]
This vid shows the wrong time being set in calendar using safari on iPhone. I am in VA. 

https://github.com/user-attachments/assets/d13a694c-4f68-4e20-934b-ee06a343cdc6

**Comment by furick1** [comment]
![IMG_4893](https://github.com/user-attachments/assets/2b7eadd0-5ded-4c24-8fee-43bcd047206b)

When using safari and adding mtg, my calendar shows the mtg at 8pm instead of 5pm. I think the fix would be either to not convert the time going into .ics start and end strings and leave IANA in or remove the IANA from the output to .ics since it’s already converted to users local time.

**Comment by tim-rohrer** [comment]
I'm converting this to a discussion per our (c4R) request to start all conversations there. If we confirm there is a bug, we'll take care of creating an issue.

**Comment by dougie-wougie** [comment]
Not related, but kinda related. I click on the Calendar in a meeting on the MGA and get the message, "Could not add an event, no calendar present." How do I get the feature to use my calendar? I am using Google Calendar.

**Reply by furick1** [reply]
Depends on what OS and web browser you are using. If Mac use safari and try. If windows try edge. Chrome on either OS isn’t configured to open the .ics file which is probably why you are getting that message.

**Reply by dougie-wougie** [reply]
I am using the Meeting Guide App on an iPhone.

**Comment by tim-rohrer** [comment]
@furick1, we've looked at the issue and concur it is likely a bug. We're focusing on the TSML UI code first . Please see [this issue](https://github.com/code4recovery/tsml-ui/issues/402) for tracking the fix.

**Comment by joshreisner** [comment]
hi @furick1 - this problem should be solved now thanks to @gkovats 🎉 

i'm seeing correct dates now when i add to calendar from the example you shared at the top

<img width="302" alt="Screenshot 2024-08-12 at 6 37 34 AM" src="https://github.com/user-attachments/assets/cbcfe4b7-7260-48a0-8a3d-9a5c0cdb1d41">

**Comment by furick1** [comment]
Confirmed fixed. Thank you @gkovats
